youtube_dl/extractor/ruleporn.py

```diff
@@ -16,29 +17,93 @@ class RulePornIE(NuevoBaseIE):
             'description': 'md5:6d28be231b981fff1981deaaa03a04d5',
             'age_limit': 18,
             'duration': 635.1,
-        }
-    }
+        }, 
+    }, {
+        'url': 'http://ruleporn.com/short-sweet-amateur-milf-sex-action/',
+        'md5': '9ec215fe7ecc19323eba42d0f16af054',
+        'info_dict': {
+            'id': '777084',
+            'display_id': 'short-sweet-amateur-milf-sex-action',
+            'ext': 'mp4',
+            'title': 'Short but sweet amateur MILF sex action',
+            'description': 'md5:a20fabf0f267839dfcde0b56a418147f',
+            'age_limit': 18,
+            'duration': 182,
+        },
+    }, {
+        'url': 'http://ruleporn.com/horny-bruentte-teen-getting-penetrated-in-a-doggy/',
+        'md5': '6f3eebefd27d1b9d28f1366d951aec56',
+        'info_dict': {
+            'id': '975925',
+            'display_id': 'horny-bruentte-teen-getting-penetrated-in-a-doggy',
+            'ext': 'mp4',
+            'title': 'Horny Brunette Teen Getting Penetrated In A Doggy',
+            'description': 'md5:2c22da523c47418e254343f9ca454758',
+            'age_limit': 18,
+            'duration': 112,
+        },
+    }]
 
     def _real_extract(self, url):
         display_id = self._match_id(url)
 
         webpage = self._download_webpage(url, display_id)
 
-        video_id = self._search_regex(
-            r'lovehomeporn\.com/embed/(\d+)', webpage, 'video id')
-
         title = self._search_regex(
-            r'<h2[^>]+title=(["\'])(?P<url>.+?)\1',
+            r'<h2[^>]+title=(["\'])\s?(?P<url>.+?)\1',
             webpage, 'title', group='url')
         description = self._html_search_meta('description', webpage)
 
-        info = self._extract_nuevo(
-            'http://lovehomeporn.com/media/nuevo/econfig.php?key=%s&rp=true' % video_id,
-            video_id)
-        info.update({
-            'display_id': display_id,
-            'title': title,
-            'description': description,
-            'age_limit': 18
-        })
+        video_url = self._search_regex(
+            r'<div\s+id=\"player\">\s+<iframe[\w\W]+?(?:src=[\"\']?(?P<url>[^\"\' ]+))', webpage, 'video url')
+
+        mobj = re.search(r'lovehomeporn\.com/embed/(\d+)', video_url)
+
+        if mobj:
+
+            video_id = mobj.group(1)
+            info = self._extract_nuevo(
+                'http://lovehomeporn.com/media/nuevo/econfig.php?key=%s&rp=true' % video_id,
+                video_id)
+
+            info.update({
+                'display_id': display_id,
+                'title': title,
+                'description': description,
+                'age_limit': 18
+            }) 
+
+        else:
+
+            video_page = self._download_webpage(video_url, display_id)
+
+            js_str = self._search_regex(
+                r'<script\s+type=\"text/javascript\">\s+var\s+item\s+=\s+({[^}]+});', video_page, 'js_str')
```

Comment:
No need to include script tag. It also won't work if JSON happen to contain inner dict.